acmw-db-app/components/AttendanceForm.jsx

```diff
@@ -53,11 +49,124 @@ const AttendanceForm = (props) => {
+    const [active, setActive] = useState(false);
+    const [value, setValue] = useState("");
+    const [label, setLabel] = useState(props.label);
+    const [predicted, setPredicted] = useState(props.predicted);
+    const [type, setType] = useState(props.type);
+  
+    const changeValue = (event) => {
+      setValue(event.target.value);
+      props.onChange(event.target.value);
+    }
+  
+    const handleKeyPress = (event) => {
+      if (predicted && predicted.startsWith(value) && event.key === "Enter") {
+        setValue(predicted);
+        props.onChange(predicted);
+      }
+    }
+
+/* APPEARANCE LOGIC
+     if not active and value (user input) is empty
+      then display placeholder white and background lighter purpur
+
+     if not active and value is non-empty
+      then display value white and background purpur
+
+     if active (it is focused bc user clicked/tabbed to it)
+      then display label bright purpur; display value dark gray; hide placeholder
+  */
+ return (
+    <div className={`${styles.field} ${active ? styles.active : ( value && styles.filled )}`}>
+      { type === "text" ?
+        ( active &&
+          predicted &&
+          predicted.startsWith(value) ?
+          <p className={styles.predicted}>{predicted}</p> :
+          ( !value && <p className={styles.placeholder}>{label}</p> ) ) :
+          ( type === "Last Name" &&
+            !active &&
+            !value &&
+            <p className={styles.placeholder}>{label}</p> )}
+      <input
+        id={id}
+        type={type}
+        value={value}
+        onChange={event => changeValue(event)}
+        onKeyPress={event => handleKeyPress(event)}
+        onFocus={() => setActive(true)}
+        onBlur={() => setActive(false)}
+      />
+      <label htmlFor={id}>
+        {label}
+      </label>
+    </div>
+  );
+}
+
+const SubmitButton = (props) => {
+
+  const [message, setMessage] = useState("");
+
+    // If event code is valid, add a database entry, and display "Submitted successfully" message
+    // Else if the event code is not valid, display "Invalid event code" message
+
+    const validateSubmitButton = async () => {
+    const regex = new RegExp(/^[a-z ,.'-]+$/i); //tests for first name
+    const regexLast = new RegExp(/^[a-z ,.'-]+\.[1-9]+$/i); //tests for last name.#
+
+    // can change the 007 to actual event codes later
+    // This would be changed to pull from a database of event codes and see if there is a match (maybe?)
+    // Sara: Actually this form might be slightly different depending on how the website is set up.
+    // If ACM-W users are able to login to their personal accounts, then we would just need the event code.
+    // This is just going off of past attendance forms.
+    
+    if ((props.eventCode == "007") && regex.test(props.firstName) && regexLast.test(props.lastName)) {
+        props.onSubmit("Success");
+        setMessage("Successfully submitted your attendance.")
```

Comment:
Are we gonna make a submitted page later? I think it kind of displays like an error message right now, but we probably don't want that. Or success can take us back to the main page or something, but this is probs for another story.

Reply:
ill make a ticket for a new submitted page